src/main/java/ps/as/BudgetApp.java

```diff
@@ -0,0 +1,84 @@
+package ps.as;
+
+import java.util.Scanner;
+
+public class BudgetApp {
+
+    Scanner scanner = new Scanner(System.in);
+
+    public void run() {
+        while (true) {
+            printMenu();
+            String option = getOption();
+            TransactionDao DAO = new TransactionDao();
```

Comment:
Lepiej utworzyć ten obiekt raz, przed pętlą, a nie przy każdym powtarzaniu pętli. Dodatkowo nazewnictwo zmiennych lokalnych to `dao`